### DISCUSSION TITLE
Is ContactListener ever called outside of PhysicsSystem::Update?

### DISCUSSION THREAD

**Discussion by zopsicle** [jrouwe/JoltPhysics#1229] [Q&A]
In Jolt Physics the contact listener is set as a property on the physics system rather than passed as an argument to Update. This suggests that it can be called outside of Update. Does that ever happen? Can I use it like the following?

```cpp
for (;;) {

    /* game code here */

    MyContactListener contactListener;
    physicsSystem.SetContactListener(&contactListener);
    physicsSystem.Update(…);
    physicsSystem.SetContactListener(nullptr);

    /* game code here */

}
```

**Comment by jrouwe** [comment]
Yes you can set the ContactListener at any time outside of PhysicsSystem::Update.

**Comment by SirLynix** [comment]
To clarify, no. The contact listener (nor any callback) will be triggered outside of PhysicsSystem::Update as no Jolt code runs. Jolt threads start and finish all their jobs during an Update call.

**Comment by zopsicle** [comment]
Thanks for the responses. :)
